File: libs/modules/cpu/src/commonMain/kotlin/com/smellouk/konitor/cpu/CpuInfo.kt
package com.smellouk.konitor.cpu

import com.smellouk.konitor.api.Info

data class CpuInfo(
    val totalUseRatio: Double,
    val appRatio: Double,
    val userRatio: Double,
    val systemRatio: Double,
    val ioWaitRatio: Double
) : Info {
    companion object {
        val INVALID = CpuInfo(-1.0, -1.0, -1.0, -1.0, -1.0)
        val UNSUPPORTED = CpuInfo(-2.0, -2.0, -2.0, -2.0, -2.0)
    }
}

File: libs/ui/flutter/lib/src/models/cpu_info.dart
class CpuInfo {
  final double totalUseRatio;
  final double appRatio;
  final double userRatio;
  final double systemRatio;
  final double ioWaitRatio;

  const CpuInfo({
    required this.totalUseRatio,
    required this.appRatio,
    required this.userRatio,
    required this.systemRatio,
    required this.ioWaitRatio,
  });

  factory CpuInfo.fromMap(Map<dynamic, dynamic> m) => CpuInfo(
        totalUseRatio: (m['totalUseRatio'] as num).toDouble(),
        appRatio: (m['appRatio'] as num).toDouble(),
        userRatio: (m['userRatio'] as num).toDouble(),
        systemRatio: (m['systemRatio'] as num).toDouble(),
        ioWaitRatio: (m['ioWaitRatio'] as num).toDouble(),
      );
}
